cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -171,26 +182,51 @@ func (sub EthereumSub) Start(completionEvent *sync.WaitGroup) {
 			completionEvent.Add(1)
 			go sub.Start(completionEvent)
 			return
+		case err := <-subHead.Err():
+			sub.Logger.Error(err.Error())
+			completionEvent.Add(1)
+			go sub.Start(completionEvent)
+			return
+		case newHead := <-heads:
+			sub.Logger.Info(fmt.Sprintf("New header %d with hash %v", newHead.Number, newHead.Hash()))
+
+			events, ok := sub.EventsBuffer.Buffer[newHead.Number]
+			if ok && events.BlockHash != newHead.Hash() {
+				// Deal with block reorg, delete all events with old hash value
+				sub.Logger.Info(fmt.Sprintf("Block reorg found old hash is %v new hash is %v", events.BlockHash, newHead.Hash()))
+				delete(sub.EventsBuffer.Buffer, newHead.Number)
+			}
+
+			// Iterate find out older enough events
+			for key, value := range sub.EventsBuffer.Buffer {
+				sub.Logger.Info(fmt.Sprintf("New header is %d stored header is %d", newHead.Number, key))
+				tmpKey := big.NewInt(30)
+				tmpKey.Add(tmpKey, key)
+				// event happened 30 blocks ago
+				if newHead.Number.Cmp(tmpKey) > 0 {
```

Comment:
If you could add docs around this process, it would be helpful in enabling others to ramp up to speed with what peggy is doing.

Reply:
Write an adr for ebrelayer to explain current ebrelayer design and improvement in the future.